R/quote.R

```diff
@@ -121,17 +130,87 @@ quote_identifier <-
 
 #' @rdname hidden_aliases
 #' @export
-setMethod("dbQuoteIdentifier", c("DBIConnection"), quote_identifier)
+setMethod("dbQuoteIdentifier", signature("DBIConnection"), quote_identifier)
 
 # Need to keep other method declarations around for now, because clients might
 # use getMethod(), see e.g. https://github.com/r-dbi/odbc/pull/149
 #' @rdname hidden_aliases
 #' @export
-setMethod("dbQuoteIdentifier", c("DBIConnection", "character"), quote_identifier)
+setMethod("dbQuoteIdentifier", signature("DBIConnection", "character"), quote_identifier)
 
 #' @rdname hidden_aliases
 #' @export
-setMethod("dbQuoteIdentifier", c("DBIConnection", "SQL"), quote_identifier)
+setMethod("dbQuoteIdentifier", signature("DBIConnection", "SQL"), quote_identifier)
+
+#' Unquote identifiers
+#'
+#' Call this method to convert a [SQL] object created by [dbQuoteIdentifier()]
+#' back to a list of [Table] objects.
+#'
+#' @param conn A subclass of [DBIConnection-class], representing
+#'   an active connection to an DBMS.
+#' @param x An [SQL] or [Table] object or character vector, or a list of such
+#'   objects, to unquote.
+#' @param ... Other arguments passed on to methods.
+#'
+#' @template methods
+#' @templateVar method_name dbUnquoteIdentifier
+#'
+#' @inherit DBItest::spec_sql_unquote_identifier return
+#' @inheritSection DBItest::spec_sql_unquote_identifier Specification
+#'
+#' @family DBIResult generics
+#' @export
+#' @examples
+#' # Unquoting allows to understand the structure of a possibly complex quoted
+#' # identifier
+#'
+#' dbUnquoteIdentifier(
+#'   ANSI(),
+#'   SQL(c("Schema"."Table", "UnqualifiedTable"))
+#' )
+#'
+#' # Character vectors are wrapped in a list
+#' dbQuoteIdentifier(
+#'   ANSI(),
+#'   c(schema = "Schema", table = "Table")
+#' )
+#'
+#' # Lists of character vectors are returned unchanged
+#' dbQuoteIdentifier(
+#'   ANSI(),
+#'   list(c(schema = "Schema", table = "Table"), "UnqualifiedTable")
+#' )
+setGeneric("dbUnquoteIdentifier",
+  def = function(conn, x, ...) standardGeneric("dbUnquoteIdentifier")
+)
+
+#' @rdname hidden_aliases
+#' @export
+setMethod("dbUnquoteIdentifier", signature("DBIConnection"), function(conn, x, ...) {
+  if (is.list(x)) {
+    return(vapply(x, dbUnquoteIdentifier, conn = conn, list(1)))
+  }
+  if (is(x, "SQL")) {
+    . <- strsplit(as.character(x), '^"|"$|"[.]"')
+    . <- lapply(., `[`, -1L)
+    split <- .
```

Comment:
I realize you likely converted this from a magrittr pipeline, but I think this would be better as

```
split <- lapply(strsplit(as.character(x), '^"|"$|"[.]"'), `[`, -1L)
```

Reply:
And actually I think it is more explicit to use `tail` than `[` here, but that is largely a matter of taste.

Reply:
I don't like nested expressions, and the dot looks like the lesser of all evils here to me, because I don't need to invent names for intermediates I don't care about.

Reply:
To be honest, I really don't like the use of `.` here. It feels so different to the style of the rest of the code.